### PR TITLE
Mentor walkthrough -- relates to issue #19

### PR DESCRIPTION
**Pull request by MissArray**
Replaced `err` with `error` in various steps.
Relates to #19

### CODE REVIEW

**Review by mattlub — CHANGES_REQUESTED** [review]
can we delete the package.json and package lock in the root?

**Comment by arrested-developer** [comment]
No PR needed as this is the mentor branch

**Comment by mattlub** [comment]
@arrested-developer I think we usually prefer having the mentor notes on master but in a folder, for ease of keeping up to date with master. (I think @eliascodes explained this better somewhere in a comment)

**Comment by eliasmalik** [comment]
https://github.com/foundersandcoders/master-reference/issues/456#issuecomment-315138262

**Comment by arrested-developer** [comment]
Thanks @eliascodes - I think the PR was opened in order to add Art's changes, but they were changes committed directly to this branch...

Reopening the PR so we can move mentor notes to the main branch (#12)

Some attention is needed first:

- [ ] this branch is 2 commits behind master, these changes need merging in
- [ ] mentor notes need to be in a 'solution' or 'mentor notes' folder
- [ ] 'empty' solution, i.e. current contents of master should be in the root folder

**Comment by mattlub** [comment]
@arrested-developer @eliascodes what about all the `steps` folders?

Should they:
- stay directly in root
- or be nested in, e.g. a `steps` folder
- or in the `mentor notes` folder?

@MissArray do you have time to finish off this PR or should I try to pick it up?

**Comment by MissArray** [comment]
@mattlub. Entirely up to you and what @arrested-developer thinks is best.
 I have already asked Michael whether he'd like one of the week-4 mentors and me to go through this repo and try to clean it up. I won't have time this week, as I'm working, but should have enough time next week in between mentoring.

**Comment by arrested-developer** [comment]
@mattlub I think the steps folders are the steps of the code-along for the mentors to follow - so should be moved inside a `solution` folder

**Comment by mattlub** [comment]
@MissArray @arrested-developer ok thanks. I have time to try to do it today so will give it a go- it's not really adding anything, just cleaning up a bit and moving files.

**Review by mattlub — APPROVED** [review]

File: public/index.js [hunk 1]
```diff
@@ -1,29 +1,27 @@
...
-
   var redBtn = document.getElementById('red-button');
   var blueBtn = document.getElementById('blue-button');
   var yellowBtn = document.getElementById('yellow-button');
   var body = document.querySelector('body');
 
   function httpGetRequest(colour) {
     var xhr = new XMLHttpRequest();
-    xhr.onreadystatechange = () => {
+    xhr.onreadystatechange = function() {
```
Comment (shiryz) [inline public/index.js:9]: The only line with es5, check the general comment

**Review by shiryz — COMMENTED** [review]
The whole workshop is written in es6 while es6 is being taught the day after...
Would be happy if a note was added about this so no one would have to change the entire ws

File: public/index.js (hunk 1 above)
Comment (m4v15) [reply public/index.js:9]: Presumably this stuff is all es5 in this file cos it's the client side JS? Like I know that's probably stupid nowadays, but it still gives a chance to talk about the difference environments this code is all running in?

**Comment by m4v15** [comment]
@shiryz I agree there needs to be a note added about es6 es5 but can it be in a seperate PR after this has been merged?

**Comment by shiryz** [comment]
@m4v15 absolutely!

**Review by shiryz — APPROVED** [review]

**Comment by m4v15** [comment]
@arrested-developer as it is so close to delivery I wanted to check with you/the mentors for this week in London whether we should merge this or leave it on a branch for now?

**Comment by MissArray** [comment]
@m4v15 Maybe merge after Monday? At the moment we know where everything is and that it works as expected.

**Comment by m4v15** [comment]
@MissArray yep, no problem - are you delivering this week in London?

**Comment by MissArray** [comment]
@m4v15, yes, @sunuwars, @DenisKent and I will be there this week.

**Comment by mattlub** [comment]
can this be merged now? @m4v15 @MissArray @eliascodes @arrested-developer

**Comment by MissArray** [comment]
@mattlub So far as I'm concerned, yes, but best to see whether the rest agree.

**Review by m4v15 — APPROVED** [review]

**Comment by m4v15** [comment]
leaving for @eliascodes to merge

**Pull request merged**